File: java/shared/src/main/java/Order.java
public class Order {
    private String itemName;
    private int amount;
    private boolean filled = false;

    public Order(String itemName, int amount) {
        this.itemName = itemName;
        this.amount = amount;
    }

    public void fill(Warehouse w) {
        if (w.hasInventory(itemName, amount)) {
            w.remove(itemName, amount);
            this.filled = true;
        }
    }

    public boolean isFilled() {
        return filled;
    }
}

File: ruby/lib/order.rb
class Order
  def initialize(item_name, amount)
    @item_name, @amount = item_name.to_s, amount.to_i
    @filled = false
  end

  def filled?
    !!@filled
  end

  def fill(warehouse)
    if warehouse.has_inventory?(@item_name, @amount)
      warehouse.remove(@item_name, @amount)
      @filled = true
    end
  end
end
